internal/controllers/objecttemplate/template_reconciler.go

```diff
@@ -304,18 +306,23 @@ func (r *templateReconciler) templateObject(
 	return nil
 }
 
-func (r *templateReconciler) getEnvironment() (map[string]any, error) {
-	env := map[string]any{}
-	packageEnvironment, err := json.Marshal(r.GetEnvironment())
+func (r *templateReconciler) getEnvironment(ctx context.Context, namespace string) (map[string]any, error) {
+	env, err := r.GetEnvironment(ctx, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get environment: %w", err)
+	}
+
+	envData := map[string]any{}
+	packageEnvironment, err := json.Marshal(env)
 	if err != nil {
-		return env, fmt.Errorf("marshaling: %w", err)
+		return envData, fmt.Errorf("marshaling: %w", err)
 	}
 
-	err = json.Unmarshal(packageEnvironment, &env)
+	err = json.Unmarshal(packageEnvironment, &envData)
```

Comment:
This looks expensive. What is the rationale behind remarshalling the environment data to/from json here?

Reply:
From a voice chat with @thetechnick:

The rationale is: Values are exposed to the package templates through maps instead of go values so that template authors can use lower-case field names to access data (like this: `.foo.bar.baz`) which mixes better with the surrounding yaml manifests of a template.


Picture this:

```yaml
# ...
spec:
  lbCount: {{ .Config.Loadbalancers.Count }}
```

vs this:

```yaml
# ...
spec:
  lbCount: {{ .config.loadbalancers.count }}
```

The second example feels more intuitive to write and reason about, especially if you also write the manifest the provides the accessed data and then are confused about specifying the data in lowercase field names and accessing the data with uppsercase names on the other side.

The json remarshal roundtrip into a `map[string]any` is a quick and dirty implementation of this.